### PR TITLE
raftstore: divide the processing logic of snapshot in ready

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

###  What have you changed?

The processing logic of snapshot in ready is a little bit different from the normal ready.
So we have lots of conditions to distinguish them which makes the code difficult to understand.
This PR try to divide these two logics to different functions and remove these conditions.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- No code

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by BusyJay** [comment]
I don't think it's a good idea, as it introduces duplicated code and split entry points. It's very easy to miss updates in both type of methods in the future.

**Comment by gengliqi** [comment]
> I don't think it's a good idea, as it introduces duplicated code and split entry points. It's very easy to miss updates in both type of methods in the future.

If there are many duplicated codes, we should extract them into a function.  As far as I can see, these two logics differ more than the same. So I think it's more clear to divide these two logics.

**Comment by BusyJay** [comment]
Then you can extract snapshot handling and append handling into different functions like:
```
fn do_snapshot_specific_job() {}
fn do_not_snapshot_specific_job() {}

fn handle_raft_ready(&self) {
    common_work();
    if snapshot() {
        do_snapshot_specific_job();
    } else {
        do_not_snapshot_specific_job();
    }
    common_work();
}
```

In my opinion, unified entry point and less duplicated code look better to me. I don't need to think twice and worry about loss of context during reading. And it's easier to update them correctly in the future.

**Comment by gengliqi** [comment]
> Then you can extract snapshot handling and append handling into different functions like:
> 
> ```
> fn do_snapshot_specific_job() {}
> fn do_not_snapshot_specific_job() {}
> 
> fn handle_raft_ready(&self) {
>     common_work();
>     if snapshot() {
>         do_snapshot_specific_job();
>     } else {
>         do_not_snapshot_specific_job();
>     }
>     common_work();
> }
> ```
> 
> In my opinion, unified entry point and less duplicated code look better to me. I don't need to think twice and worry about loss of context during reading. And it's easier to update them correctly in the future.

I modified some code and rename several unclear function names. Could you please review it again?

File: components/raftstore/src/store/peer.rs [hunk 1]
```diff
@@ -1248,22 +1267,11 @@ impl Peer {
...
-                ctx.need_flush_trans = true;
-            }
-        }
+        fail_point!("raft_before_follower_send");
+        self.pending_messages = mem::replace(&mut ready.messages, vec![]);
 
-        if apply_snap_result.is_some() {
-            self.activate(ctx);
+        self.activate(ctx);
+        {
```
Comment (BusyJay) [inline components/raftstore/src/store/peer.rs]: The scope is unnecessary.

**Pull request closed without merging**